packages/model-viewer/src/features/scene-graph/material.ts

```diff
@@ -154,4 +155,78 @@ export class Material extends ThreeDOMElement implements MaterialInterface {
     (this[$sourceObject] as GLTFMaterial).emissiveFactor = rgb;
     this[$onUpdate]();
   }
+
+  setAlphaCutoff(cutoff: number): void {
+    cutoff = Math.min(1.0, cutoff);
+    for (const material of this[$correlatedObjects] as
+         Set<MeshStandardMaterial>) {
+      material.alphaTest = cutoff;
+      material.needsUpdate = true;
+    }
+    (this[$sourceObject] as GLTFMaterial).alphaCutoff = cutoff;
+    this[$onUpdate]();
+  }
+
+  getAlphaCutoff(): number {
+    return (this[$sourceObject] as GLTFMaterial).alphaCutoff ??
+        this[$correlatedObjects]!.values().next().value.alphaTest;
+  }
+
+  setDoubleSided(doubleSided: boolean): void {
+    for (const material of this[$correlatedObjects] as
+         Set<MeshStandardMaterial>) {
+      // When double-sided is disabled gltf spec dictates that Back-Face culling
+      // must be disabled, in three.js parlance that would mean FrontSide
+      // rendering only.
+      // https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#double-sided
+      material.side = doubleSided ? DoubleSide : FrontSide;
+      material.needsUpdate = true;
+    }
+    (this[$sourceObject] as GLTFMaterial).doubleSided = doubleSided;
+    this[$onUpdate]();
+  }
+
+  getDoubleSided(): boolean {
+    return (this[$sourceObject] as GLTFMaterial).doubleSided ??
+        this[$correlatedObjects]!.values().next().value.side === DoubleSide;
+  }
+
+  setAlphaMode(alphaMode: AlphaMode): void {
+    for (const material of this[$correlatedObjects] as
+         Set<MeshStandardMaterial>) {
+      material.blending = NormalBlending;
+      if (alphaMode === 'OPAQUE') {
+        this.setAlphaCutoff(ALPHA_CUTOFF_OPAQUE);
+      } else if (alphaMode === `BLEND`) {
+        this.setAlphaCutoff(ALPHA_CUTOFF_BLEND);
+      } else {
+        // MASK mode.
+        if (this.getAlphaCutoff() < 0) {
+          this.setAlphaCutoff(ALPHA_CUTOFF_GLTF_DEFAULT);
+        }
+      }
+      material.needsUpdate = true;
+    }
+
+    (this[$sourceObject] as GLTFMaterial).alphaMode = alphaMode;
+    this[$onUpdate]();
+  }
+
+  getAlphaMode(): AlphaMode {
+    if ((this[$sourceObject] as GLTFMaterial).alphaMode !== undefined) {
+      return (this[$sourceObject] as GLTFMaterial).alphaMode!;
+    }
+
+    const blendMode = this[$correlatedObjects]!.values().next().value.blending;
+
+    if (blendMode === NormalBlending && this.getAlphaCutoff() >= 0) {
+      return 'BLEND';
+    } else if (blendMode === NoBlending && this.getAlphaCutoff() > 0) {
```

Comment:
There's something funny here about checking `NoBlending`, but not setting it. Might mean we need another test too.

Reply:
Hmm, I realize now that we can just return 'OPAQUE' if `alphaMode` is not defined. This is because the spec says `alphaMode` default is OPAQUE, so everything else has to have a defined value.